File: python_sdk/src/reality_capture/specifications/gaussian_splats.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class GaussianSplatsInputs(BaseModel):
    scene: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene to process.")
    splats_reference: Optional[str] = Field(default=None,
                                            description="Reality data ID (cloud) or local path (on-premise) of the Gaussian Splats Reference.",
                                            alias="splatsReference")
    region_of_interest: Optional[str] = Field(description="Path to region of interest file (bucket path in cloud, local path on-premise)",
                                              alias="regionOfInterest",
                                              default=None)
    crs_data: Optional[str] = Field(default=None,
                                    description="Path to CRS data file (bucket path in cloud, local path on-premise).",
                                    alias="crsData")
    preset: Optional[str] = Field(default=None, description="Path to preset")


class GaussianSplatsOutputs(BaseModel):
    splats: Optional[str] = Field(default=None, description="Reality data ID (cloud) or local path (on-premise) of Gaussian Splats.")
    splats_reference: Optional[str] = Field(default=None,
                                            description="Reality data ID (cloud) or local path (on-premise) of the Gaussian Splats Reference.",
                                            alias="splatsReference")


class GaussianSplatsOutputsCreate(Enum):
    SPLATS = "splats"
    SPLATS_REFERENCE = "splatsReference"


class GSFormat(str, Enum):
    PLY = "PLY"
    SPZ = "SPZ"
    THREED_TILES = "3DTiles"
    THREED_TILES_LOD = "3DTilesLOD"


class GSQuality(str, Enum):
    MEDIUM = "Medium"
    STANDARD = "Standard"
    HIGH = "High"


class GaussianSplatsOptions(BaseModel):
    export_crs: Optional[str] = Field(default=None, description="CRS for the exported gaussian splats",
                                      alias="exportCrs")
    export_format: Optional[GSFormat] = Field(default=None, description="Format of the exported Gaussian Splats",
                                              alias="exportFormat")
    reference_quality: Optional[GSQuality] = Field(default=None,
                                                   description="Quality to use to "
                                                               "generate the Splats Reference",
                                                   alias="referenceQuality")
    reference_tile_size: Optional[float] = Field(default=None, description="Tile size for the Splats Reference",
                                                 alias="referenceTileSize")


class GaussianSplatsSpecifications(BaseModel):
    inputs: GaussianSplatsInputs = Field(description="Inputs")
    outputs: GaussianSplatsOutputs = Field(description="Outputs")
    options: Optional[GaussianSplatsOptions] = Field(default=None, description="Options")


class GaussianSplatsSpecificationsCreate(BaseModel):
    inputs: GaussianSplatsInputs = Field(description="Inputs")
    outputs: list[GaussianSplatsOutputsCreate] = Field(description="Outputs")
    options: Optional[GaussianSplatsOptions] = Field(default=None, description="Options")

File: typescript/packages/reality-capture/src/specifications/gaussian_splats.ts
import { z } from "zod";

export const GaussianSplatsInputsSchema = z.object({
    scene: z.string().describe("Reality data ID of ContextScene to process."),
    splatsReference: z.string().optional().describe("Reality data ID of the Gaussian Splats Reference.")
});
export type GaussianSplatsInputs = z.infer<typeof GaussianSplatsInputsSchema>;

export const GaussianSplatsOutputsSchema = z.object({
    splats: z.string().optional().describe("Reality data ID of Gaussian Splats."),
    splatsReference: z.string().optional().describe("Reality data ID of the Gaussian Splats Reference.")
});
export type GaussianSplatsOutputs = z.infer<typeof GaussianSplatsOutputsSchema>;

export enum GaussianSplatsOutputsCreate {
  SPLATS = "Splats",
  SPLATS_REFERENCE = "SplatsReference"
}

export enum GSFormat {
  SPZ = "SPZ",
  THREED_TILES = "3DTiles",
  PLY = "PLY"
}

export enum GSQuality {
  MEDIUM = "Medium",
  STANDARD = "Standard",
  HIGH = "High"
}

export const GaussianSplatsOptionsSchema = z.object({
    export_format: z.nativeEnum(GSFormat).optional().describe("Format of the Gaussian Splats"),
    reference_quality: z.nativeEnum(GSQuality).optional().describe("Quality to use to generate the Splats Reference"),
    reference_tile_size: z.number().optional().describe("Tile size for the Splats Reference")
});
export type GaussianSplatsOptions = z.infer<typeof GaussianSplatsOptionsSchema>;

export const GaussianSplatsSpecificationsSchema = z.object({
    inputs: GaussianSplatsInputsSchema.describe("Inputs"),
    outputs: GaussianSplatsOutputsSchema.describe("Outputs"),
    options: GaussianSplatsOptionsSchema.optional().describe("Options")
});
export type GaussianSplatsSpecifications = z.infer<typeof GaussianSplatsSpecificationsSchema>;

export const GaussianSplatsSpecificationsCreateSchema = z.object({
    inputs: GaussianSplatsInputsSchema.describe("Inputs"),
    outputs: z.array(z.nativeEnum(GaussianSplatsOutputsCreate)).describe("Outputs"),
    options: GaussianSplatsOptionsSchema.optional().describe("Options")
});
export type GaussianSplatsSpecificationsCreate = z.infer<typeof GaussianSplatsSpecificationsCreateSchema>;